be/src/exec/vectorized/jdbc_scanner.cpp

```diff
@@ -0,0 +1,485 @@
+    int max_len = slot_desc->type().len;
+    if (cpp_val.size() > max_len) {
+        return Status::DataQualityError(
+                fmt::format("Value length exceeds limit on column[{}], max length is [{}], value is [{}]",
+                            slot_desc->col_name(), max_len, cpp_val));
+    }
+    switch (slot_desc->type().type) {
+    case TYPE_CHAR:
+    case TYPE_VARCHAR: {
+        Slice val(cpp_val);
+        _append_data<TYPE_VARCHAR, Slice>(column, val);
+        break;
+    }
+    default: {
+        DCHECK(false) << "unreachable path, unknown type:" << slot_desc->type().type;
+        return Status::InternalError(fmt::format("unknown type {}", slot_desc->type().type));
+    }
+    }
+    return Status::OK();
+}
+
+template <typename CppType>
+Status JDBCScanner::_append_value_from_result(jobject jval, std::function<CppType(jobject)> get_value_func,
+                                              SlotDescriptor* slot_desc, Column* column) {
+    PROCESS_NULL_VALUE(jval, column)
+
+#define CHECK_DATA_OVERFLOW(val, min_val, max_val)                                                                  \
+    if (val > max_val || val < min_val) {                                                                           \
+        return Status::DataQualityError(                                                                            \
+                fmt::format("Data out of range on column[{}], invalid value is [{}]", slot_desc->col_name(), val)); \
+    }
+
+    CppType cpp_val = get_value_func(jval);
+    switch (slot_desc->type().type) {
+    case TYPE_INT: {
+        CHECK_DATA_OVERFLOW(cpp_val, INT_MIN, INT_MAX)
+        _append_data<TYPE_INT, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_TINYINT: {
+        CHECK_DATA_OVERFLOW(cpp_val, INT8_MIN, INT8_MAX)
+        _append_data<TYPE_TINYINT, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_SMALLINT: {
+        CHECK_DATA_OVERFLOW(cpp_val, INT16_MIN, INT16_MAX)
+        _append_data<TYPE_SMALLINT, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_BIGINT: {
+        CHECK_DATA_OVERFLOW(cpp_val, INT64_MIN, INT64_MAX)
+        _append_data<TYPE_BIGINT, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_DOUBLE: {
+        _append_data<TYPE_DOUBLE, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_FLOAT: {
+        _append_data<TYPE_FLOAT, CppType>(column, cpp_val);
+        break;
+    }
+    case TYPE_BOOLEAN: {
+        _append_data<TYPE_BOOLEAN, CppType>(column, cpp_val);
+        break;
+    }
+    default: {
+        DCHECK(false) << "unknown type:" << slot_desc->type().type;
+        return Status::InternalError(fmt::format("unknown type {}", slot_desc->type().type));
+    }
+    }
+    return Status::OK();
+}
+
+Status JDBCScanner::_fill_chunk(jobject jchunk, ChunkPtr* chunk) {
+    auto& helper = JVMFunctionHelper::getInstance();
+    jobject first_column = helper.list_get(jchunk, 0);
```

Comment:
we need free first_column

Reply:
we could use _env->pushFrame and _env->popFrame() instead of deleteLocalRef

Reply:
> we need free first_column

fixed